src/dev/tslint/rules/noImportZonesRule.js

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+const path = require('path');
+const Lint = require('tslint');
+const mm = require('micromatch');
+const { findImports, ImportKind } = require('tsutils');
+
+function traverseToTopFolder(src, pattern) {
+  while(src && mm([src], pattern).length) {
```

Comment:
I think it's okay to have a local fork of the `no-import-zones rule`, we can port some logic to **eslint** as well (if we want ofc).